src/Microsoft.Health.Fhir.Core/Features/Definition/ISearchParameterDefinitionManager.cs

```diff
@@ -70,17 +71,17 @@ public interface ISearchParameterDefinitionManager
         /// <returns>The search parameter type.</returns>
         SearchParamType GetSearchParameterType(SearchParameterInfo searchParameter, int? componentIndex);
 
-        /// <summary>
-        /// Updates the existing resource type - search parameter hash mapping with the given new values.
-        /// </summary>
-        /// <param name="updatedSearchParamHashMap">Dictionary containing resource type to search parameter hash values</param>
-        public void UpdateSearchParameterHashMap(Dictionary<string, string> updatedSearchParamHashMap);
-
         /// <summary>
         /// Gets the hash of the current search parameters that are supported for the given resource type.
         /// </summary>
         /// <param name="resourceType">Resource type for which we need the hash of search parameters.</param>
         /// <returns>A string representing a hash of the search parameters.</returns>
         public string GetSearchParameterHashForResourceType(string resourceType);
+
+        /// <summary>
+        /// Allows addition of a new search parameters at runtime
+        /// </summary>
+        /// <param name="searchParamBundle">A bundle containing SearchParameter resources</param>
+        public void AddNewSearchParameters(BundleWrapper searchParamBundle);
```

Comment:
>BundleWrapper searchParamBundle [](start = 43, length = 31)

I'm a bit puzzled by fact it's a bundle.
Is there are reason why it's not just `List<SearchParameterInfo>` ?

Reply:
When we read in the static file of the search parameters from the spec, they come in as a bundle.  Also, when we query for SearchParameter resources from the datastore, they should come as a bundle.  We could change it, read the bundle and get the list of entries.

Reply:
I'm just looking on content of this file, and it's called `SearchParameter`DefinitionManager and in all other methods it operates and produce `SearchParameterInfo` (with exception of Dictionary<string,string>, string and SearchParameterType).
If I need to perform singular Get I will get back `SearchParameterInfo` if I do all , I get back list of `SearchParameterInfo`

If I look into code which uses this method  we also wrap single resource into bundle, so later it can be unpacked which I find a bit convoluted.

I think it would be nice to have separation between how things passed to us and how we operate on them.  It's better to operate on one entity and have converters before we pass information, than support multiple things at once. 

---
In reply to: [532759584](https://github.com/microsoft/fhir-server/pull/1449#discussion_r532759584) [](ancestors = 532759584)

Reply:
As discussed, reason behind this method accepting bundle, is because this is shared among 3 different specifications code and there is no way to pick right SearchParam. SearchParamInfo is something we use internally to represent SearchParam and it has limited coverage compared to any implmentation of SearchParam so we can't use it either.

---
In reply to: [532826560](https://github.com/microsoft/fhir-server/pull/1449#discussion_r532826560) [](ancestors = 532826560,532759584)

Reply:
After refactoring to move the call to addsearchparameters out of the filter, it was easier to also move away from using Bundle and switch to ITypedElement.